arbor/mechanisms/SK_E2.mod

```diff
@@ -0,0 +1,49 @@
+: SK-type calcium-activated potassium current
+: Reference : Kohler et al. 1996
+
+NEURON {
+    SUFFIX SK_E2
+    USEION k READ ek WRITE ik
+    USEION ca READ cai
+    RANGE gSK_E2bar
+}
+
+UNITS {
+    (mV) = (millivolt)
+    (mA) = (milliamp)
+    (mM) = (milli/liter)
+}
+
+PARAMETER {
+    gSK_E2bar = 0.000001 (mho/cm2)
+    zTau      = 1        (ms)
+}
+
+STATE {
+    z FROM 0 TO 1
+}
+
+BREAKPOINT {
+    SOLVE states METHOD cnexp
+    ik = gSK_E2bar*z*(v - ek)
+}
+
+DERIVATIVE states {
+    LOCAL ca, zInf
+    ca = cai
+    if (ca < 1e-7) {
```

Comment:
what if `ca==-1e-7`? This would blow up. It isn't a realistic value for `cai`, but ... numerics.

What we really want is to set `zInf=0` when `ca` is very close to zero, so what about:

```
if (cai<1e-7) {
    zInf = 0
}
else {
    zInf = 1/(1 + (0.00043/ca)^4.8)
}
```

Reply:
That way we don't need a shadow variable to work around the hack of directly modifying `cai`.